include/rbs/rbs_location.h

```diff
@@ -12,9 +12,9 @@ typedef struct rbs_location {
     rbs_loc_children *children;
 } rbs_location_t;
 
-void rbs_loc_alloc_children(rbs_location_t *loc, int size);
-void rbs_loc_add_required_child(rbs_location_t *loc, rbs_constant_id_t name, range r);
-void rbs_loc_add_optional_child(rbs_location_t *loc, rbs_constant_id_t name, range r);
+void rbs_loc_alloc_children(rbs_allocator_t *allocator, rbs_location_t *loc, int size);
+void rbs_loc_add_required_child(rbs_allocator_t *allocator, rbs_location_t *loc, rbs_constant_id_t name, range r);
+void rbs_loc_add_optional_child(rbs_allocator_t *allocator, rbs_location_t *loc, rbs_constant_id_t name, range r);
```

Comment:
You can leave out the parameter name for brevity in the header, where the type makes it self-explaining

```suggestion
void rbs_loc_alloc_children(rbs_allocator_t *, rbs_location_t *loc, int size);
void rbs_loc_add_required_child(rbs_allocator_t *, rbs_location_t *loc, rbs_constant_id_t name, range r);
void rbs_loc_add_optional_child(rbs_allocator_t *, rbs_location_t *loc, rbs_constant_id_t name, range r);
```

(Arguably many of the others could also been removed, but I kept those similar to what it was before. For new code using allocators, I've been using a bare `bs_allocator_t *`)